Fix HA containers denials with rhel8 and podman

mbaldessari opened the pull request:
We get currently two type of denials when starting HA containers on
OSP15:
A) Rabbitmq creates and deletes a bmtp file
denied { remove_name } for  pid=538235 comm="python" name="btmp" dev="vda2" ino=73787843 scontext=system_u:system_r:container_t:s0:c208,c543 tcontext=system_u:object_r:cluster_var_log_t:s0 tclass=dir permissive=1
denied { unlink } for  pid=538235 comm="python" name="btmp" dev="vda2" ino=73787843 scontext=system_u:system_r:container_t:s0:c208,c543 tcontext=system_u:object_r:cluster_var_log_t:s0 tclass=file permissive=1
denied { create } for  pid=538235 comm="python" name="btmp" scontext=system_u:system_r:container_t:s0:c208,c543 tcontext=system_u:object_r:cluster_var_log_t:s0 tclass=file permissive=1
denied { ioctl } for  pid=538235 comm="python" path="/var/log/btmp" dev="vda2" ino=73787840 ioctlcmd=0x5401 scontext=system_u:system_r:container_t:s0:c208,c543 tcontext=system_u:object_r:cluster_var_log_t:s0 tclass=file permissive=1
denied { setattr } for  pid=538235 comm="python" name="btmp" dev="vda2" ino=73787840 scontext=system_u:system_r:container_t:s0:c208,c543 tcontext=system_u:object_r:cluster_var_log_t:s0 tclass=file permissive=1
denied { add_name } for  pid=46749 comm="python" name="btmp" scontext=system_u:system_r:container_t:s0:c527,c559 tcontext=system_u:object_r:cluster_var_log_t:s0 tclass=dir permissive=1
denied { write open } for  pid=46749 comm="python" path="/var/log/btmp" dev="vda2" ino=67376554 scontext=system_u:system_r:container_t:s0:c527,c559 tcontext=system_u:object_r:cluster_var_log_t:s0 tclass=file permissive=1

B) Kolla needs to open, read, lock and create folders
denied { add_name } for  pid=41771 comm="mkdir" name="kolla" scontext=system_u:system_r:container_t:s0:c296,c382 tcontext=system_u:object_r:cluster_var_log_t:s0 tclass=dir permissive=1
denied { create } for  pid=41771 comm="mkdir" name="kolla" scontext=system_u:system_r:container_t:s0:c296,c382 tcontext=system_u:object_r:cluster_var_log_t:s0 tclass=dir permissive=1
denied { create } for  pid=42078 comm="su" name="lastlog" scontext=system_u:system_r:container_t:s0:c296,c382 tcontext=system_u:object_r:cluster_var_log_t:s0 tclass=file permissive=1
denied { lock } for  pid=52961 comm="su" path="/var/log/lastlog" dev="vda2" ino=73669296 scontext=system_u:system_r:container_t:s0:c383,c514 tcontext=system_u:object_r:cluster_var_log_t:s0 tclass=file permissive=1
denied { read } for  pid=73801 comm="mkdir" name="kolla" dev="vda2" ino=38118009 scontext=system_u:system_r:container_t:s0:c670,c1013 tcontext=system_u:object_r:cluster_var_log_t:s0 tclass=dir permissive=1
denied { read write open } for  pid=52961 comm="su" path="/var/log/lastlog" dev="vda2" ino=73669296 scontext=system_u:system_r:container_t:s0:c383,c514 tcontext=system_u:object_r:cluster_var_log_t:s0 tclass=file permissive=1
denied { write } for  pid=46749 comm="python" name="rabbitmq-bundle-0" dev="vda2" ino=67376550 scontext=system_u:system_r:container_t:s0:c527,c559 tcontext=system_u:object_r:cluster_var_log_t:s0 tclass=dir permissive=1
denied { write } for  pid=73801 comm="mkdir" name="ovn-dbs-bundle-0" dev="vda2" ino=86202322 scontext=system_u:system_r:container_t:s0:c670,c1013 tcontext=system_u:object_r:cluster_var_log_t:s0 tclass=dir permissive=1

We tested this patch and subsequently restart all HA containers:
haproxy, rabbit, galera, redis and ovn-dbs

And could not observe any denials any longer.

Closes-Bug: rhbz#1692325

Co-Authored-By: Damien Ciabrini <dciabrin@redhat.com>

jpichon commented:
Awesome, thank you for submitting a fix!

It looks like one of the types used is missing from "gen_require()" at the top of the file. Adding "type cluster_var_log_t;" in there should resolve that.

Generally this can be simplified by using macros similar to line 15:
manage_files_pattern(container_t, cluster_var_log_t, cluster_var_log_t);
manage_dirs_pattern(container_t, cluster_var_log_t, cluster_var_log_t);

Also, would it be possible to add a tests/bzXXXXXX file that contains the (full) AVCs mentioned in the description? That way we can both check it works after deploying the RPM and prevent future regressions. Thank you.

mbaldessari commented:
Thanks for the feedback @jpichon. We're clearly selinux noobs, let me fix this PR up with your input ;)

mbaldessari commented:
Ok I hope I addressed your feedback. I have retested the change when using manage_*_patterns and added my successful testing result to the commit message. Also added the testcase to the tests/folders (although I have not run any test check)
